lib/mpi/service.rb

```diff
@@ -309,12 +310,17 @@ def message_user_attributes(user_identity, search_type, orch_search: false)
         ssn: user_identity.ssn,
         gender: user_identity.gender
       }
-      MPI::Messages::FindProfileMessage.new(
-        profile,
-        search_type: search_type,
-        orch_search: orch_search,
-        edipi: orch_search == true ? user_identity.edipi : nil
-      ).to_xml
+
+      if user_identity.ssn.nil? && user_identity.address.present?
+        MPI::Messages::FindProfileMessageWithAddress.new(profile.merge(address: user_identity.address)).to_xml
```

Comment:
I think I'd prefer this up a function, actually (at least the logic on whether to FindProfile with address or with ssn), since the rest of the attributes acting as query-switches are in the `create_profile_message` function. Something like:
`message_user_attributes_address(user_identity, search_type) if user_identity.ssn.nil && user_identity.address.present?`